Update WebAssembly Document (WASM.md).

jplevyak opened the pull request:
Signed-off-by: John Plevyak <jplevyak@gmail.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

PiotrSikora commented:
Why did you close #80? In any case, my comments from there apply here.

duderino commented on WASM.md:
Technically correct, but I can say later versions currently do not work

jplevyak replied:
https://github.com/envoyproxy/envoy-wasm/pull/83

I am making up to 1.38.39 work.

jplevyak replied:
#83 has been merged.

jplevyak commented:
Addressed the comments here.  #80 I though was on the wrong repo because github says:

envoyproxy:master from jplevyak:wasm-docs

which doesn't specify the envoy-wasm repo.  Turns out you have to mouse over... but why ?

PiotrSikora commented:
Note: This doesn't generate the way you think it does, see: https://github.com/envoyproxy/envoy-wasm/blob/26fa937a9bc713fa17deb57c6bcea512966312db/WASM.md

jplevyak commented:
Fixed the formatting.  For the record, adding two spaces at the end of the line to force a line break.

jplevyak commented:
Except that check_format.py doesn't like that.  Sigh.

PiotrSikora commented on WASM.md:
Replace this line with dependency name alone (i.e. `glib2.0`).

jplevyak replied:
Done.

PiotrSikora commented on WASM.md:
`emscripten` is a dependency for regenerating test modules, isn't it?

jplevyak replied:
It is also a dependency for building your own WebAssembly module, so it is should be in the main section.

PiotrSikora commented on WASM.md:
`protobuf` is a dependency for regenerating test modules, isn't it?

jplevyak replied:
It is also a dependency for building your own WebAssembly module which uses any protos (e.g. gRPC which is part of the standard API), so it should be included in the main section.
